atlp-rwanda/winners-c8-bn-be
# #182300803 Authentication check middlewares-

## Auth check middlewares storyID:182300803
## What does this PR do?
1.  It is a middleware that check if the user is authenticated. and assign user with response if Authorized elsewhere display an error
 ## More descriptions about this task.
1. Middleware Authentication to verify token
2. Helper to generate token
3. Routes to work within your controllers
## How should it be manually tested?
1.  Clone this repo into your local computer
2. Create a `.env` file referring to the `.env.example` file in the root folder and provide values for those keys.
3. Run `npm` install for installing all dependencies.
4. Run `npm run dev` to start server
5.  Check by any routes  with token and without token to see if there is difference by providing error  
## Screenshoot if appropriate?
1.  N/A
## What are the relevant pivotal tracker stories?
Story type: `feature(feat)`
StoryID:  `182300803`
